homeassistant/components/fibaro.py

```diff
@@ -0,0 +1,360 @@
+        _LOGGER.warning("Not sure how to setValue: %s "
+                        "(available actions: %s)", str(self.ha_id),
+                        str(self.fibaro_device.actions))
+
+    def set_level(self, level):
+        """Set the level of Fibaro device."""
+        self.action("setValue", level)
+        if 'value' in self.fibaro_device.properties:
+            self.fibaro_device.properties.value = level
+        if 'brightness' in self.fibaro_device.properties:
+            self.fibaro_device.properties.brightness = level
+
+    def set_color(self, red, green, blue, white):
+        """Set the tilt level of Fibaro device."""
+        color_str = "{},{},{},{}".format(int(red), int(green),
+                                         int(blue), int(white))
+        self.fibaro_device.properties.color = color_str
+        self.action("setColor", str(int(red)), str(int(green)),
+                    str(int(blue)), str(int(white)))
+
+    def action(self, cmd, *args):
+        """Perform an action on the Fibaro HC."""
+        if cmd in self.fibaro_device.actions:
+            getattr(self.fibaro_device, cmd)(*args)
+            _LOGGER.debug("-> %s.%s%s called", str(self.ha_id),
+                          str(cmd), str(args))
+        else:
+            self.dont_know_message(cmd)
+
+    @property
+    def hidden(self) -> bool:
+        """Return True if the entity should be hidden from UIs."""
+        return self.fibaro_device.visible is False
+
+    @property
+    def current_power_w(self):
+        """Return the current power usage in W."""
+        if 'power' in self.fibaro_device.properties:
+            power = self.fibaro_device.properties.power
+            if power:
+                return convert(power, float, 0.0)
+        else:
+            return 0
+
+    @property
+    def current_binary_state(self):
+        """Return the current binary state."""
+        if self.fibaro_device.properties.value == 'false':
+            return False
+        if self.fibaro_device.properties.value == 'true' or \
+                int(self.fibaro_device.properties.value) > 0:
+            return True
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Get polling requirement from fibaro device."""
+        return False
+
+    def update(self):
+        """Call to update state."""
+        pass
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attr = {}
+
+        try:
+            if 'battery' in self.fibaro_device.interfaces:
+                attr[ATTR_BATTERY_LEVEL] = \
+                    int(self.fibaro_device.properties.batteryLevel)
+            if 'fibaroAlarmArm' in self.fibaro_device.interfaces:
+                attr[ATTR_ARMED] = True if \
+                    self.fibaro_device.properties.armed else False
```

Comment:
`attr[ATTR_ARMED] = bool(self.fibaro_device.properties.armed)`

Reply:
Actually, you can even avoid the `bool()` cast if `armed` is itself a boolean :)

Reply:
My trouble is that I haven't fully tested this functionality, so I'll need to test and see to be sure